fs/copy.py

```diff
@@ -286,9 +334,10 @@ def dst():
     from ._bulk import Copier
 
     with src() as _src_fs, dst() as _dst_fs:
-        with _src_fs.lock(), _dst_fs.lock():
-            _thread_safe = is_thread_safe(_src_fs, _dst_fs)
-            with Copier(num_workers=workers if _thread_safe else 0) as copier:
+        _thread_safe = is_thread_safe(_src_fs, _dst_fs)
+        copier = Copier(num_workers=workers if _thread_safe else 0)
+        with copier:
+            with _src_fs.lock(), _dst_fs.lock():
```

Comment:
Any reason to create the copier before locking instead of after like before?

Reply:
I had to change the logic of `Copier` because of some internal locking mechanics. But I just noticed that that change might actually have caused all workers to work sequentially, rather than in parallel. So I reverted most of my changes in `fs._bulk`. The modification times are now updated when closing the `Copier` instance, after all tasks have finished.